### PR TITLE
Merge 2.1 master

### PR DESCRIPTION
**Pull request by SamirBoulil**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added Behats                      | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/CatalogBundle/EventSubscriber/SaveFamilyVariantOnFamilyUpdateSubscriber.php [hunk 1]
```diff
@@ -26,14 +27,22 @@ class SaveFamilyVariantOnFamilyUpdateSubscriber implements EventSubscriberInterf
...
+     * @param ValidatorInterface          $validator
+     * @param BulkSaverInterface          $familyVariantSaver
+     * @param BulkObjectDetacherInterface $objectDetacher
      */
-    public function __construct(ValidatorInterface $validator, BulkSaverInterface $familyVariantSaver)
-    {
+    public function __construct(
+        ValidatorInterface $validator,
+        BulkSaverInterface $familyVariantSaver,
+        BulkObjectDetacherInterface $objectDetacher = null
```
Comment (ahocquard) [inline src/Pim/Bundle/CatalogBundle/EventSubscriber/SaveFamilyVariantOnFamilyUpdateSubscriber.php]: delete `= null `

File: src/Pim/Bundle/CatalogBundle/EventSubscriber/SaveFamilyVariantOnFamilyUpdateSubscriber.php [hunk 2]
```diff
@@ -75,6 +84,10 @@ public function validateAndSaveFamilyVariants(GenericEvent $event): void
 
         $this->familyVariantSaver->saveAll($validFamilyVariants);
 
+        if (null !== $this->objectDetacher) {
```
Comment (ahocquard) [inline src/Pim/Bundle/CatalogBundle/EventSubscriber/SaveFamilyVariantOnFamilyUpdateSubscriber.php]: delete condition

File: src/Pim/Bundle/EnrichBundle/Normalizer/ProductNormalizer.php [hunk 3]
```diff
@@ -191,6 +197,12 @@ public function normalize($product, $format = null, array $context = [])
 
         $scopeCode = $context['channel'] ?? null;
 
+        $incompleteValues = [];
+        // TODO: remove this check on master
+        if (null !== $this->incompleteValuesNormalizer) {
```
Comment (ahocquard) [inline src/Pim/Bundle/EnrichBundle/Normalizer/ProductNormalizer.php]: remove. Typically what I would like a tag for this kin of comment

File: src/Pim/Bundle/EnrichBundle/Normalizer/ProductNormalizer.php [hunk 4]
```diff
@@ -143,7 +147,8 @@ public function __construct(
         EntityWithFamilyValuesFillerInterface $productValuesFiller,
         EntityWithFamilyVariantAttributesProvider $attributesProvider,
         VariantNavigationNormalizer $navigationNormalizer,
-        AscendantCategoriesInterface $ascendantCategoriesQuery
+        AscendantCategoriesInterface $ascendantCategoriesQuery,
+        NormalizerInterface $incompleteValuesNormalizer = null
```
Comment (ahocquard) [inline src/Pim/Bundle/EnrichBundle/Normalizer/ProductNormalizer.php]: delete `= null `

File: src/Akeneo/Component/StorageUtils/Cache/CacheClearerInterface.php [hunk 5]
```diff
@@ -5,6 +5,9 @@
 /**
  * Clears the cache
  *
+ * TODO This interface is not a "cache clearer", it's an entity manager clearer, was not renamed to avoid BC.
```
Comment (ahocquard) [inline src/Akeneo/Component/StorageUtils/Cache/CacheClearerInterface.php]: Should be renamed

File: src/Pim/Bundle/EnrichBundle/Normalizer/ProductNormalizer.php (hunk 3 above)
Comment (SamirBoulil) [reply src/Pim/Bundle/EnrichBundle/Normalizer/ProductNormalizer.php]: ok

**Pull request merged**